src/web/releases.rs

```diff
@@ -69,26 +69,36 @@ pub(crate) fn get_releases(conn: &mut Client, page: i64, limit: i64, order: Orde
 
     // WARNING: it is _crucial_ that this always be hard-coded and NEVER be user input
     let (ordering, filter_failed): (&'static str, _) = match order {
-        Order::ReleaseTime => ("releases.release_time", false),
-        Order::GithubStars => ("repositories.stars", false),
-        Order::RecentFailures => ("releases.release_time", true),
-        Order::FailuresByGithubStars => ("repositories.stars", true),
+        Order::ReleaseTime => ("release_time", false),
+        Order::GithubStars => ("stars", false),
+        Order::RecentFailures => ("release_time", true),
+        Order::FailuresByGithubStars => ("stars", true),
     };
     let query = format!(
-        "SELECT crates.name,
-            releases.version,
-            releases.description,
-            releases.target_name,
-            releases.release_time,
-            releases.rustdoc_status,
-            repositories.stars
-        FROM crates
-        INNER JOIN releases ON crates.latest_version_id = releases.id
-        LEFT JOIN repositories ON releases.repository_id = repositories.id
-        WHERE
-            ((NOT $3) OR (releases.build_status = FALSE AND releases.is_library = TRUE)) 
-            AND {0} IS NOT NULL
-
+        "SELECT name,
+            version,
+            description,
+            target_name,
+            release_time,
+            rustdoc_status,
+            stars
+        FROM (
+            SELECT crates.name,
+                releases.version,
+                releases.description,
+                releases.target_name,
+                releases.release_time,
+                releases.rustdoc_status,
+                repositories.stars,
+                rank() OVER (PARTITION BY repository_id ORDER BY release_time DESC) AS rank
```

Comment:
this will filter all releases without repository (where `releases.repository_id` is `NULL`) except the most recent one.